### PR TITLE
extras: add a script to build heketi containers using buildah

### PR DESCRIPTION
**Pull request by phlogistonjohn**
I have been successfully using this script for a while to
build containers for test purposes. It has the ability to
build heketi from source (similar to extras/docker/fromsource)
but as an added bonus has an option for using using existing
heketi binaries and can thus cut down on the time needed to
build a heketi container.
All modes are based on the 'buildah' tool which is a daemon-less
alternative to 'docker build'.

### CODE REVIEW

**Comment by phlogistonjohn** [comment]
This is a simple(ish) standalone script. Any opinions?

File: extras/buildah/build.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    heketi_cli_bin = pjoin(heketi_dir, 'client/cli/go/heketi-cli')
+    shutil.copy(heketi_bin, outdir)
+    shutil.copy(heketi_cli_bin, outdir)
+    buildah('umount', bc)
+
+    buildah('rm', bc)
+
+
+def build_container(sources):
+    hc = buildah_from('fedora:27')
```
Comment (nixpanic) [inline extras/buildah/build.py]: This could probably be `fedora:latest`?

File: extras/buildah/build.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    return buildah('from', src, redirect=True)
+
+
+def build_binaries(outdir):
+    pkgs = 'glide golang git make mercurial'
+    hdir = '/build/src/github.com/heketi/heketi'
+    heketi_branch = 'master'
+    heketi_url = 'https://github.com/heketi/heketi.git'
+
+    bc = buildah_from('fedora:27')
```
Comment (nixpanic) [inline extras/buildah/build.py]: I assume `fedora:27` to get a functional version of glide? A comment and maybe reference to the BZ would be  good.

File: extras/buildah/build.py (hunk 1 above)
Comment (phlogistonjohn) [reply extras/buildah/build.py]: There was an issue using fedora 28 IIRC. What I should do is move that to a top-level variable and make it possible to swap the base image easier for testing.

File: extras/buildah/build.py (hunk 2 above)
Comment (phlogistonjohn) [reply extras/buildah/build.py]: Yes, good idea.

**Comment by phlogistonjohn** [comment]
Updated.

**Pull request merged**